slave-base/contrib/bin/run-jnlp-client

```diff
@@ -36,6 +36,7 @@ echo "OPENSHIFT_JENKINS_JVM_ARCH='${OPENSHIFT_JENKINS_JVM_ARCH}', CONTAINER_MEMO
 
 # Configure the slave image
 source /usr/local/bin/configure-slave
+source /usr/local/bin/configure-agent
```

Comment:
isn't this going to cause file not found errors?  (might not fail the script but it's going to look ugly).

how about wrapping each call if a test to see if the file exists and only source it if so?

Reply:
there were no error messages that I saw when running the images manually, looking at pod logs, so I kept it vanilla

Reply:
i'm skeptical
```
$ cat /tmp/t.sh 
#!/bin/sh

source foo
```

```
$ /tmp/t.sh 
/tmp/t.sh: line 3: source: foo: file not found
```


slave-base includes a dummy configure-slave:
https://github.com/openshift/jenkins/tree/master/slave-base/contrib/bin

so you won't see it for your new agent images.

but you will see the error on the old slave images that don't have a configure-agent, unless you add a dummy one.

Reply:
ah didn't catch what slave-base did ... ok that would explain things

OK ... so from a bash pespective and the if test, do you mean:

```
if ! source /usr/local/bin/configure-agent ; then
   source /usr/local/bin/configure-slave
fi
```
or some such?

Reply:
i meant 
`
if [ -e /usr/local/bin/configure-slave ]; then
  source /usr/local/bin/configure-slave
fi
`

but i think i'd rather you just introduce the dummy file in the base image like we did for configure-slave.